reports/graphs/feeding_amounts.py

```diff
@@ -14,23 +14,57 @@ def feeding_amounts(instances):
     :param instances: a QuerySet of Feeding instances.
     :returns: a tuple of the the graph's html and javascript.
     """
-    totals = {}
+    feeding_types = [
+            'breast milk',
+            'formula',
+            'fortified breast milk',
+            'solid food',
+        ]
+    feeding_types_desc = [
+            'Breast milk',
+            'Formula',
+            'Fortified breast milk',
+            'Solid food',
+        ]
+    total_idx = len(feeding_types) + 1  # +1 for aggregate total
+    totals_list = list()
+    for i in range(total_idx):
+        totals_list.append({})
     for instance in instances:
         end = timezone.localtime(instance.end)
         date = end.date()
-        if date not in totals.keys():
-            totals[date] = 0
-        totals[date] += instance.amount or 0
-
-    amounts = [round(amount, 2) for amount in totals.values()]
-    trace = go.Bar(
-        name=_('Total feeding amount'),
-        x=list(totals.keys()),
-        y=amounts,
+        if date not in totals_list[total_idx-1].keys():
+            for item in totals_list:
+                item[date] = 0
+        feeding_idx = feeding_types.index(instance.type)
+        totals_list[feeding_idx][date] += instance.amount or 0
+        totals_list[total_idx-1][date] += instance.amount or 0
+    zeros = [0 for a in totals_list[total_idx-1].values()]
+
+    # sum each feeding type for graph
+    amounts_array = []
+    for i in range(total_idx):
+        amounts_array.append([round(a, 2) for a in totals_list[i].values()])
+
+    traces = []
+    for i in range(total_idx-1):
+        traces.append(go.Bar(
+            name=feeding_types_desc[i],
+            x=list(totals_list[total_idx-1].keys()),
+            y=amounts_array[i],
+            text=amounts_array[i]
+
+        ))
+
+    traces.append(go.Bar(
+        name=_('Total'),
+        x=list(totals_list[total_idx-1].keys()),
+        y=zeros,
         hoverinfo='text',
         textposition='outside',
-        text=amounts
-    )
+        text=amounts_array[total_idx-1],
+        showlegend=False
+    ))
 
     layout_args = utils.default_graph_layout_options()
     layout_args['title'] = _('<b>Total Feeding Amounts</b>')
```

Comment:
Lets update the title of this graph to "Total Feeding Amount by Type". We'll need to do this elsewhere as well.

Reply:
I changed as suggested but it seems like this should somehow come from the translations. I'm not sure how to make that happen.

Reply:
@alzyee yep no worries -- that is what `_` does. It's a weird Python convention: https://github.com/babybuddy/babybuddy/blob/master/reports/graphs/feeding_amounts.py#L3. Translations will need to be updated to pick up the change but I'll do that separately.